Scrabble.Bot/Scrabble.fs

```diff
@@ -43,22 +43,24 @@ module State =
     // but it could, potentially, keep track of other useful
     // information, such as number of players, player turn, etc.
 
-    type state =
-        { board: Parser.board
-          dict: ScrabbleUtil.Dictionary.Dict
-          playerNumber: uint32
-          hand: MultiSet.MultiSet<uint32> }
-
-    let mkState b d pn h =
-        { board = b
-          dict = d
-          playerNumber = pn
-          hand = h }
-
-    let board st = st.board
-    let dict st = st.dict
-    let playerNumber st = st.playerNumber
-    let hand st = st.hand
+    type state = {
+        board         : Parser.board
+        dict          : ScrabbleUtil.Dictionary.Dict
+        playerId  : uint32
+        hand          : MultiSet.MultiSet<uint32>
+        players       : array<bool>
+        playerTurn    : uint32
+    }
+
+    let mkState b d pn h pl pt = {board = b; dict = d;  playerId = pn; hand = h; players = pl; playerTurn = pt}
+    let removePlayer st player = st.players[player-1] <- false

```

Comment:
Is it a good idea to mutate the array values?

Reply:
I have changed it to lists.